src/Symfony/Component/Translation/Bridge/Phrase/PhraseProviderFactory.php

```diff
@@ -60,14 +70,46 @@ public function create(Dsn $dsn): ProviderInterface
             ],
         ]);
 
-        $readConfig = ReadConfig::fromDsn($dsn);
-        $writeConfig = WriteConfig::fromDsn($dsn);
+        $readConfig = $this->readConfigFromDsn($dsn);
+        $writeConfig = $this->writeConfigFromDsn($dsn);
 
-        return new PhraseProvider($client, $this->logger, $this->loader, $this->xliffFileDumper, $this->cache, $this->defaultLocale, $endpoint, $readConfig, $writeConfig);
+        return new PhraseProvider($client, $this->logger, $this->loader, $this->xliffFileDumper, $this->cache, $this->defaultLocale, $endpoint, $readConfig, $writeConfig, $this->isFallbackLocaleEnabled($dsn));
     }
 
     protected function getSupportedSchemes(): array
     {
         return ['phrase'];
     }
+
+    private function isFallbackLocaleEnabled(Dsn $dsn): bool
+    {
+        $options = $dsn->getOptions()['read'] ?? [];
+
+        return filter_var($options['fallback_locale_enabled'] ?? false, \FILTER_VALIDATE_BOOL);
+    }
+
+    private function readConfigFromDsn(Dsn $dsn): array
+    {
+        $options = $dsn->getOptions()['read'] ?? [];
+
+        // enforce empty translations when fallback locale is enabled
+        if (true === $this->isFallbackLocaleEnabled($dsn)) {
```

Comment:
we don't compare to booleans unless another type can be returned (not the case here)